lib/PreCleanup.ml

```diff
@@ -76,6 +76,91 @@ let remove_array_eq =
       super#visit_DFunction (n_cgs, 0) cc flags n_cgs n t lid bs e
   end
 
+let expression_of_cg (cg : cg) =
+  match cg with
+  | CgConst n -> Krml.Helpers.mk_sizet (int_of_string (snd n))
+  | CgVar _ -> failwith "non-const length in the Arr<T,C>"
+
+let expand_slice_to_array =
```

Comment:
Ok so I'm now reviewing this. I think what would help is if you showed the AST before and the AST after, in concrete syntax. Also, what is the original type of slice_to_array?